src/map/instance.cpp

```diff
@@ -809,352 +827,415 @@ enum e_instance_enter instance_enter(struct map_session_data *sd, unsigned short
+			return false;
+		}
+		if (y == -1) {
+			ShowWarning("instance_read_db_sub: Missing instance enter y location for instance %d in \"%s\", skipping.\n", instance_id, source.c_str());
+			return false;
 		}
-	}
 
-	if (isNew) {
-		uidb_put(InstanceDB, id, db);
-		strdb_uiput(InstanceNameDB, StringBuf_Value(db->name), id);
+		entry->enter.map = m;
+		entry->enter.x = x;
+		entry->enter.y = y;
+	} catch(...) {
+		yaml_invalid_warning("instance_read_db_sub: Instance definition with invalid enter location field in '" CL_WHITE "%s" CL_RESET "', skipping.\n", node, source);
+		return false;
 	}
-	return true;
-}
 
-/**
- * Free InstanceDB single entry
- * @param db Instance Db entry
- **/
-static bool instance_db_free_sub(struct instance_db *db) {
-	if (!db)
-		return 1;
-	StringBuf_Free(db->name);
-	StringBuf_Free(db->enter.mapname);
-	if (db->maplist_count) {
-		uint8 i;
-		for (i = 0; i < db->maplist_count; i++)
-			StringBuf_Free(db->maplist[i]);
-		aFree(db->maplist);
+	if (node["AdditionalMaps"]) {
+		try {
+			const YAML::Node map_list = node["AdditionalMaps"];
+
+			if (map_list.IsSequence()) {
+				for (uint16 i = 0; i < map_list.size(); i++) {
+					std::string mapname;
+					int16 m;
+
+					mapname = map_list[i].as<std::string>();
+					m = map_mapname2mapid(mapname.c_str());
+
+					if (!m) {
+						ShowWarning("instance_read_db_sub: Invalid additional map for instance %d in \"%s\", skipping.\n", instance_id, source.c_str());
+						return false;
+					}
+					entry->maplist.push_back(m);
+				}
+			} else
+				ShowWarning("instance_read_db_sub: Invalid additional map format for instance %d in \"%s\".\n", instance_id, source.c_str());
+		} catch (...) {
+			yaml_invalid_warning("instance_read_db_sub: Instance definition with invalid additional map field in '" CL_WHITE "%s" CL_RESET "', skipping.\n", node, source);
+			return false;
+		}
 	}
-	aFree(db);
-	return 0;
-}
 
-/**
- * Free InstanceDB entries
- **/
-static int instance_db_free(DBKey key, DBData *data, va_list ap) {
-	struct instance_db *db = (struct instance_db *)db_data2ptr(data);
-	return instance_db_free_sub(db);
+	return true;
 }
 
 /**
- * Read instance_db.txt files
- **/
+ * Loads instances from the instance database
+ */
 void instance_readdb(void) {
-	const char* filename[] = { DBPATH"instance_db.txt", "import/instance_db.txt"};
-	int f;
+	std::vector<std::string> directories = { std::string(db_path) + "/" + std::string(DBPATH),  std::string(db_path) + "/" + std::string(DBIMPORT) + "/" };
+	static const std::string file_name("instance_db.yml");
+
+	for (const auto &directory : directories) {
+		std::string current_file = directory + file_name;
+		YAML::Node config;
+		int count = 0;
+
+		try {
+			config = YAML::LoadFile(current_file);
+		} catch (...) {
+			ShowError("Cannot read '" CL_WHITE "%s" CL_RESET "'.\n", current_file.c_str());
+			return;
```

Comment:
not continue instead ?
I'd found we just discard that file and continue reading the rest